// exp-3/src/App.js
import { BrowserRouter, Routes, Route, NavLink, Link } from "react-router-dom";
import "./App.css";

/* ===== Components ===== */

// Home
function Home() {
  return (
    <div className="page">
      <section className="hero">
        <div className="hero-text">
          <p className="eyebrow">Running Experiment 3</p>
          <h2>Build a multi-page SPA that feels like a product launch.</h2>
          <p className="subtext">
            This lab upgrades simple routing into a cinematic UI: bold typography,
            layered depth, and content that tells a story across every page.
          </p>
          <div className="hero-actions">
            <Link className="btn primary" to="/dashboard">
              Explore Dashboard
            </Link>
            <Link className="btn ghost" to="/profile">
              View Profile
            </Link>
          </div>
          <div className="stat-row">
            <div className="stat">
              <span className="stat-value">5</span>
              <span className="stat-label">Routes</span>
            </div>
            <div className="stat">
              <span className="stat-value">12</span>
              <span className="stat-label">Components</span>
            </div>
            <div className="stat">
              <span className="stat-value">1</span>
              <span className="stat-label">Vision</span>
            </div>
          </div>
        </div>
        <div className="hero-card">
          <div className="card-glow" />
          <h3>Launch Checklist</h3>
          <div className="checklist">
            <div className="check-item">
              <span>Navigation</span>
              <span className="tag success">Ready</span>
            </div>
            <div className="check-item">
              <span>Page Content</span>
              <span className="tag success">Polished</span>
            </div>
            <div className="check-item">
              <span>UI Theme</span>
              <span className="tag info">Energized</span>
            </div>
            <div className="check-item">
              <span>Animations</span>
              <span className="tag warning">Subtle</span>
            </div>
          </div>
          <p className="muted">
            Focus on clarity, contrast, and flow. Each route owns its own
            narrative.
          </p>
        </div>
      </section>

      <section className="grid">
        <div className="panel">
          <h3>Experiments Inside</h3>
          <p>
            Route structure: Home, About, Contact, Dashboard, Profile, and a
            dedicated 404 page.
          </p>
        </div>
        <div className="panel">
          <h3>Design System</h3>
          <p>
            A layered palette, soft glows, and a display font to keep the UI
            memorable.
          </p>
        </div>
        <div className="panel">
          <h3>Interaction</h3>
          <p>
            Animated entrances, hovering energy, and crisp readability across
            devices.
          </p>
        </div>
      </section>

      <section className="timeline">
        <h3>Experiment Flow</h3>
        <div className="timeline-row">
          <div className="step">
            <span className="step-dot" />
            <div>
              <h4>Route Setup</h4>
              <p>Map URLs to clear, descriptive pages.</p>
            </div>
          </div>
          <div className="step">
            <span className="step-dot" />
            <div>
              <h4>Content Crafting</h4>
              <p>Give each page its own purpose and tone.</p>
            </div>
          </div>
          <div className="step">
            <span className="step-dot" />
            <div>
              <h4>UI Polish</h4>
              <p>Layer depth, motion, and contrast.</p>
            </div>
          </div>
        </div>
      </section>
    </div>
  );
}

// About
function About() {
  return (
    <div className="page">
      <div className="split">
        <div>
          <h2>About This Experiment</h2>
          <p className="subtext">
            We are turning routing practice into a portfolio-grade interface.
            The goal is to show navigation, layout, and content hierarchy with
            a modern visual language.
          </p>
          <div className="badge-row">
            <span className="badge">React Router</span>
            <span className="badge">UI Systems</span>
            <span className="badge">Responsive</span>
            <span className="badge">Motion</span>
          </div>
        </div>
        <div className="panel highlight">
          <h3>Why It Works</h3>
          <p>
            Clear navigation, intentional typography, and content sections that
            feel alive. Every screen has a role and a voice.
          </p>
          <div className="pill-row">
            <span className="pill">Clarity</span>
            <span className="pill">Depth</span>
            <span className="pill">Consistency</span>
          </div>
        </div>
      </div>

      <section className="story">
        <h3>Milestones</h3>
        <div className="milestones">
          <div className="milestone">
            <h4>Foundation</h4>
            <p>Set routes, layout shell, and navigation hierarchy.</p>
          </div>
          <div className="milestone">
            <h4>Experience</h4>
            <p>Design hero sections and content grids for clarity.</p>
          </div>
          <div className="milestone">
            <h4>Launch</h4>
            <p>Apply motion and final polish to elevate the UI.</p>
          </div>
        </div>
      </section>
    </div>
  );
}

// Contact
function Contact() {
  return (
    <div className="page">
      <div className="split">
        <div>
          <h2>Contact Studio</h2>
          <p className="subtext">
            Share feedback, report findings, or request an upgrade. This lab is
            built for iteration and refinement.
          </p>
          <div className="contact-card">
            <div>
              <p className="label">Email</p>
              <p className="value">lab@reactrouter.dev</p>
            </div>
            <div>
              <p className="label">Studio</p>
              <p className="value">FSD-2 Experience Bay</p>
            </div>
            <div>
              <p className="label">Availability</p>
              <p className="value">Mon-Fri, 10 AM - 6 PM</p>
            </div>
          </div>
        </div>
        <form className="panel form">
          <h3>Send a Message</h3>
          <label>
            Your Name
            <input type="text" placeholder="Enter full name" />
          </label>
          <label>
            Project Goals
            <input type="text" placeholder="What should we build next-" />
          </label>
          <label>
            Notes
            <textarea rows="4" placeholder="Share context or requirements" />
          </label>
          <button className="btn primary" type="button">
            Submit
          </button>
        </form>
      </div>
    </div>
  );
}

// Dashboard
function Dashboard() {
  return (
    <div className="page">
      <h2>Dashboard</h2>
      <p className="subtext">
        Track the experiment status, page metrics, and design quality in one
        place.
      </p>
      <section className="grid metrics">
        <div className="panel metric">
          <p className="label">Route Health</p>
          <h3>98%</h3>
          <p className="muted">Clean navigation and zero dead links.</p>
        </div>
        <div className="panel metric">
          <p className="label">UI Polish</p>
          <h3>4.9 / 5</h3>
          <p className="muted">Bold visual identity and strong hierarchy.</p>
        </div>
        <div className="panel metric">
          <p className="label">Engagement</p>
          <h3>2.3x</h3>
          <p className="muted">Users explore multiple routes.</p>
        </div>
      </section>

      <section className="panel chart">
        <div>
          <h3>Weekly Momentum</h3>
          <p className="muted">
            A visual pulse of interactions, page views, and retention.
          </p>
        </div>
        <div className="sparkline">
          <span />
          <span />
          <span />
          <span />
          <span />
        </div>
      </section>

      <section className="panel activity">
        <h3>Latest Activity</h3>
        <div className="activity-item">
          <span className="dot" />
          <div>
            <p>New hero layout applied to Home.</p>
            <p className="muted">2 minutes ago</p>
          </div>
        </div>
        <div className="activity-item">
          <span className="dot" />
          <div>
            <p>Contact form content refined.</p>
            <p className="muted">25 minutes ago</p>
          </div>
        </div>
        <div className="activity-item">
          <span className="dot" />
          <div>
            <p>Dashboard insights rebalanced.</p>
            <p className="muted">1 hour ago</p>
          </div>
        </div>
      </section>
    </div>
  );
}

// Profile
function Profile() {
  return (
    <div className="page">
      <section className="profile-hero">
        <div>
          <p className="eyebrow">Creator Profile</p>
          <h2>FSD-2 Lab Designer</h2>
          <p className="subtext">
            Building immersive interfaces, one route at a time. Focused on
            clarity, boldness, and a memorable user experience.
          </p>
        </div>
        <div className="profile-card">
          <div className="avatar">FD</div>
          <div>
            <h3>Parth</h3>
            <p className="muted">UI Explorer</p>
            <div className="pill-row">
              <span className="pill">React</span>
              <span className="pill">Design</span>
              <span className="pill">Motion</span>
            </div>
          </div>
        </div>
      </section>

      <section className="grid">
        <div className="panel">
          <h3>Core Skills</h3>
          <p>Routing, layout systems, component structure, and UI flow.</p>
        </div>
        <div className="panel">
          <h3>Focus for This Week</h3>
          <p>Polish spacing, refine typography, and keep content purposeful.</p>
        </div>
        <div className="panel">
          <h3>Next Goal</h3>
          <p>Integrate data APIs and advanced transitions.</p>
        </div>
      </section>
    </div>
  );
}

// Not Found
function NotFound() {
  return (
    <div className="page not-found">
      <h2>404</h2>
      <p className="subtext">
        This route is off the map. Let's get you back to the launchpad.
      </p>
      <Link className="btn primary" to="/">
        Return Home
      </Link>
    </div>
  );
}

/* ===== App ===== */
function App() {
  return (
    <BrowserRouter>
      <div className="app-shell">
        <nav className="navbar">
          <div className="brand">
            <span className="brand-mark" />
            <h1>React Router Lab</h1>
          </div>
          <div className="links">
            <NavLink to="/" end>
              Home
            </NavLink>
            <NavLink to="/about">About</NavLink>
            <NavLink to="/contact">Contact</NavLink>
            <NavLink to="/dashboard">Dashboard</NavLink>
            <NavLink to="/profile">Profile</NavLink>
          </div>
        </nav>

      {/* Routes */}
        <main className="content">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/about" element={<About />} />
            <Route path="/contact" element={<Contact />} />
            <Route path="/dashboard" element={<Dashboard />} />
            <Route path="/profile" element={<Profile />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </main>
      </div>
    </BrowserRouter>
  );
}

export default App;
